### PR TITLE
[ENG-2375] Adding docs around the new /reactors/:id/react endpoint

### PR DESCRIPTION
**Pull request by dhudec**
## Description

- Adding new `reactors/:id/react` endpoint docs
- Removing source token type
- deprecating atomic react endpoints

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

image
image
image

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: source/includes/api-reference/_reactor_formulas.md [hunk 1]
```diff
@@ -4,62 +4,87 @@ Reactor formulas give you the ability to pre-configure custom integrations to se
...
+## Reactor Formula Types
 
+| Type       | Description                                                                                |
+|------------|--------------------------------------------------------------------------------------------|
+| `official` | Official formulas that are built and supported by Basis Theory and its authorized partners |
+| `private`  | Private formulas which are only available to your Tenant                                   |
 
 <h2 id="reactor-formulas-create-reactor-formula">Create Reactor Formula <span class="beta menu">BETA</span></h2>
 
+TODO: describe request parameters in more detail. Object notation, type conversions,
```
Comment (koalaty-code) [inline source/includes/api-reference/_reactor_formulas.md:85]: Can we address this todo or remove it?

File: source/includes/api-reference/_reactors.md [hunk 2]
```diff
@@ -406,10 +406,214 @@ Delete a Reactor by ID in the Tenant.
...
+| `id`      | true     | *uuid* | `null`  | The ID of the Reactor |
+
+### Request Parameters
+| Parameter | Required | Type     | Default | Description                                                                                                                                                                                                   |
+|-----------|----------|----------|---------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `args`    | false    | *object* | `null`  | Arguments to provide to the reactor. These arguments must match the declared [request parameters](#reactor-formulas-reactor-formula-object-reactor-formula-request-parameter-object) for the reactor formula. |
+
+### Response
+
+Returns a [React Response](#reactors-invoke-a-reactor-reactor-response-object) if the Atomic Card was reacted successfully. Returns [an error](#errors) if the Atomic Card failed to react.
```
Comment (koalaty-code) [inline source/includes/api-reference/_reactors.md:497]: This sentence should be agnostic to any specific token or atomic type.

File: source/includes/api-reference/_reactors.md [hunk 3]
```diff
@@ -406,10 +406,214 @@ Delete a Reactor by ID in the Tenant.
...
+All required `request_parameters` must be included within the `args` on each request.
+If a request parameter was defined with `optional: false` and a value for this parameter is not provided within `args`, the request will be rejected with a 400 error.
+
+Each argument must match the `type` on the corresponding request parameter, and if the type does not match, auto casting to the declared type will be attempted. 
+For example, say a request parameter is declared with type `number` but the string value `"123"` is provided. This value can be cast to a number, so the reactor would receive the numeric value `123`. 
+However, if the string value `"non-numeric"` were provided, a 400 error would be returned because the value cannot be cast to a number.
+
+### Detokenization
+
+In order to use tokenized data within a reactor, the `args` parameter may contain token interpolation patterns of the form `{{tokenId}}`.
```
Comment (koalaty-code) [inline source/includes/api-reference/_reactors.md:551]:
```suggestion
In order to use tokenized data within a reactor, the `args` parameter may contain token interpolation patterns of the form `{{<tokenId>}}`.
```
Should we use a syntax to clearly indicate that `tokenId` is a placeholder here?

File: source/includes/api-reference/_reactors.md [hunk 4]
```diff
@@ -406,10 +406,214 @@ Delete a Reactor by ID in the Tenant.
...
+}
+```
+
+Validation is performed on the resulting request after token interpolation, so several required request parameters may be supplied by interpolating
+a single complex token that contains several of the request parameters.
+
+At most, 100 tokens may be detokenized within a single reactor request.
+
+<aside class="notice">
+  <span>For more detailed examples about the various ways to Invoke a Reactor, check out our [How to Invoke a Reactor](#todo) guide.</span>
```
Comment (koalaty-code) [inline source/includes/api-reference/_reactors.md:618]: This links to `#todo`. Is this guide coming soon or do we need to remove the link?

File: source/includes/api-reference/_reactor_formulas.md (hunk 1 above)
Comment (dhudec) [reply source/includes/api-reference/_reactor_formulas.md:85]: Oops... removed it!

File: source/includes/api-reference/_reactors.md (hunk 4 above)
Comment (dhudec) [reply source/includes/api-reference/_reactors.md:618]: It's coming next... I'm not sure what the url is yet. I can remove the link and add it later.

File: source/includes/api-reference/_atomic_banks.md [hunk 5]
```diff
@@ -584,22 +588,22 @@ React with an Atomic Bank by ID.
...
+| Parameter            | Required | Type   | Default | Description                                                                                                        |
+|----------------------|----------|--------|---------|--------------------------------------------------------------------------------------------------------------------|
+| `reactor_id`         | true     | *uuid* | `null`  | The ID of the reactor                                                                                              |
+| `request_parameters` | false    | *map*  | `null`  | A key-value map of [request parameter](#reactor-formula-request-parameter-object) names and values for the reactor |
+| `metadata`           | false    | *map*  | `null`  | (Deprecated) A key-value map of non-sensitive data to be associated with any created tokens                        |
 
 ### Response
 
 Returns a [React Response](#atomic-banks-react-with-an-atomic-bank-react-response-object) if the Atomic Bank was reacted successfully. Returns [an error](#errors) if the Atomic Bank failed to react.
```
Comment (koalaty-code) [inline source/includes/api-reference/_atomic_banks.md:599]: The `[React Response]` anchor has changed to `atomic-banks-react-with-an-atomic-bank-deprecated-...`. Do we want to update that or change the anchor to the new response object definition under `Invoke a Reactor`?

File: source/includes/api-reference/_atomic_cards.md [hunk 6]
```diff
@@ -518,22 +523,22 @@ React with an Atomic Card by ID.
...
+| Parameter            | Required | Type   | Default | Description                                                                                                        |
+|----------------------|----------|--------|---------|--------------------------------------------------------------------------------------------------------------------|
+| `reactor_id`         | true     | *uuid* | `null`  | The ID of the reactor                                                                                              |
+| `request_parameters` | false    | *map*  | `null`  | A key-value map of [request parameter](#reactor-formula-request-parameter-object) names and values for the reactor |
+| `metadata`           | false    | *map*  | `null`  | (Deprecated) A key-value map of non-sensitive data to be associated with any created tokens                        |
 
 ### Response
 
 Returns a [React Response](#atomic-cards-react-with-an-atomic-card-react-response-object) if the Atomic Card was reacted successfully. Returns [an error](#errors) if the Atomic Card failed to react.
```
Comment (koalaty-code) [inline source/includes/api-reference/_atomic_cards.md:534]: The `[React Response]` anchor has changed to `atomic-card-react-with-an-atomic-card-deprecated-...`. Do we want to update that or change the anchor to the new response object definition under `Invoke a Reactor`?

File: source/includes/api-reference/_atomic_banks.md (hunk 5 above)
Comment (dhudec) [reply source/includes/api-reference/_atomic_banks.md]: Linked to the Invoke a Reactor section - the other deprecated link was too ugly to use

File: source/includes/api-reference/_atomic_cards.md (hunk 6 above)
Comment (dhudec) [reply source/includes/api-reference/_atomic_cards.md]: Same as above, linked to the Invoke a Reactor section

**Review by koalaty-code — APPROVED** [review]

**Pull request merged**